semaphore.h

```diff
@@ -3,18 +3,21 @@
 
 using namespace std;
 #include <mutex>
+#include <atomic>
+#include <condition_variable>
 
 class Semaphore
 {
 public:
     Semaphore(int initialValue);
 
-    void Increment();       // Signal()
-    void Decrement();       // Wait()
+    void Signal();       // Increment
+    void Wait();         // Decrement
 
 private:
-    int mSemaphoreValue;
+    std::atomic<int> mSemaphoreValue;
```

Comment:
I have found in similar examples, that such values are declared as atomic. Why do you think its usage does not make sense ?

Reply:
I will mark this conversation as resolved, since the suggestion has already been mentioned above.

Reply:
It does not make sense because the value is only accessed or modified when the mutex is locked. So it does not make sense to add the costs associated with atomic since the mutex guarantees that the value will be consistent.